libs/common/src/types/types-helper.ts

```diff
@@ -7,3 +7,20 @@ export type Prettify<T> = {
 } & {};
 
 export type SafeExclude<T extends string, K extends T> = Exclude<T, K>;
+
+/**
+ * Converts all property types of an object to
+ * `unknown`
+ *
+ * @example
+ * type Body = {
+ *   id: number;
+ *   name: string;
+ * };
+ *
+ * type PreValidationBody = Unknown<Body>;
+ * //   ^? { id: unknown; name: unknown }
+ */
+export type Unknown<T> = {
```

Comment:
Will this right now only do a shallow copy to unknown e.g. what if I have a type of:

`{
  a: string,
  b: {
    c: string;
    d: string:
  },
}`
? Tbh that sounds like exactly what we want if we don't know the type that's coming into the API in the request :D
Something about the name doesn't feel descriptive enough but the best i've thought of is 'unknownise' which is just worse :D

Reply:
The name was based on typescript utility types like `Readonly` which just makes all properties `readonly`. Similarly, `Readonly` only does a shallow copy of the type:
```typescript
// lib.es5.d.ts
/**
 * Make all properties in T readonly
 */
type Readonly<T> = {
    readonly [P in keyof T]: T[P];
};
```
The usual pattern I've seen in typescript is that the standard utility type is shallow, and you would extend the type to `DeepReadonly` (for instance) if you want it to be deep.

Naming wise, I agree that `Unknown` isn't clear and I was hoping a better name would come to me 😅. Maybe `WithUnknownProperties`?